repository/src/main/java/org/alfresco/service/cmr/repository/ContentService.java

```diff
@@ -170,4 +173,62 @@ public ContentWriter getWriter(NodeRef nodeRef, QName propertyQName, boolean upd
      */
     @Auditable(parameters = {"nodeRef", "expiresAt"})
     public DirectAccessUrl getDirectAccessUrl(NodeRef nodeRef, Date expiresAt);
+
+    /**
+     * Checks whether or not the current {@link ContentService} supports the provided {@link Set} storage classes
+     *
+     * @param storageClasses The storage classes that will be checked whether or not are supported
+     * @return true if the storage classes are supported, false otherwise.
+     */
+    default boolean isStorageClassesSupported(Set<String> storageClasses)
+    {
+        return false;
+    }
+
+    /**
+     * @return Returns the complete {@link Set} of supported storage classes by this {@link ContentService}
+     */
+    default Set<String> getSupportedStorageClasses()
+    {
+        return Collections.emptySet();
+    }
+
+    /**
+     * Updates the storage class for a {@link NodeRef}
+     *
+     * @param nodeRef The ref of the node that will have its storage classes updated
+     * @param storageClasses The new storage classes
+     * @param parameters extra parameters
+     */
+    default void updateStorageClasses(NodeRef nodeRef, Set<String> storageClasses, Map<String, Object> parameters)
+    {
+
```

Comment:
We should throw an exception if this isn't implemented and someone tries to use it.

Reply:
I haven't got there with the code yet but noted.